### PR TITLE
[Doc] Added explanations for documentation landing page

### PR DESCRIPTION
**Pull request by D-2-Ed**
Doc updates and navigational support to improve user experience.

### Motivation
Improve user experience by adding "paths" and providing context.

### Modifications

Context updates for each content block. Describes where the user should start.

### Verifying this change
This is a doc change and does not require CI checks or test coverage.

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [ ] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [X] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Comment by D-2-Ed** [comment]
@Huanli-Meng - Can you approve this PR?

File: site2/docs/about.md [hunk 1]
```diff
@@ -10,21 +10,27 @@ import BlockLink from "@site/src/components/BlockLink";
...
+This portal holds a variety of topics, tutorials, guides, and reference material to help you work with Pulsar. 
 
-If you’re an experienced coder, review this page to learn the easiest way to access the specific content you’re looking for.
+## Choose your path....
+Select one of the content blocks below to begin your Pulsar journey. If you ...
+* Are new to Pulsar, start with **About Pulsar** to learn about features and concepts.
+* Want to jump to the quickstart, select **Get Started**.
+* Are an operator responsible for architecting and supporting Apache Pulsar. start with **Install, Deploy, Upgrade**.
+* Are a developer who wants to master Apache Pulsar, select **Pulsar for Developers**. 
+* If you just want to try out Pulsar, select **How To** for access to the "hello world" tutorial.
```
Comment (Huanli-Meng) [inline site2/docs/about.md:24]:
```suggestion
* Want to try out Pulsar, select **How To** for access to the "hello world" tutorial.
```

File: site2/docs/about.md [hunk 2]
```diff
@@ -10,21 +10,27 @@ import BlockLink from "@site/src/components/BlockLink";
...
+* If you are an experienced Pulsar coder looking for API, metrics, or configuration documentation, go to **Reference**.
```
Comment (Huanli-Meng) [inline site2/docs/about.md:25]:
```suggestion
* An experienced Pulsar coder looking for API, metrics, or configuration documentation, go to **Reference**. 
```

**Review by Huanli-Meng — APPROVED** [review]
Overall LGTM. Only two minor comments.

**Comment by Huanli-Meng** [comment]
> @Huanli-Meng - Can you approve this PR?

@D-2-Ed Done. PTAL, thanks.

**Comment by tisonkun** [comment]
@D-2-Ed you may tag this PR as `doc` instead of `doc-complete` :)

**Review by tisonkun — APPROVED** [review]
The content change looks good to me.

Generally, we will post a preview of the page after changes, you can try to provide one. This patch is trivial and I can approve it as is.

You can refer to [this page](https://github.com/apache/pulsar-site/blob/760f7b47f437267a3c9d71d14c73e99ba32bd1ec/site2/website-next/README.md) about how to generate a preview.

**Comment by tisonkun** [comment]
I think we can merge this patch with one more committer's approval.

cc @Anonymitaet @dave2wave

**Comment by D-2-Ed** [comment]
@Anonymitaet - can you approve? Thanks.

**Comment by Huanli-Meng** [comment]
@D-2-Ed could you please help rebase your branch? Thanks

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**